src/sinks/datadog/logs.rs

```diff
@@ -48,11 +50,13 @@ pub struct DatadogLogsConfig {
 #[derive(Clone)]
 pub struct DatadogLogsJsonService {
     config: DatadogLogsConfig,
+    uri: String,
```

Comment:
I'm guessing you separated out the `uri` to avoid calling `config.get_uri` on each request? If so, could you add a comment to that effect? It makes sense, but I was initially confused why we wouldn't just fetch the URI from the `config` when needed.

Looking at the `DatadogLogsJsonService` (and `DatadogLogsTextService`) it seems like only really the URI and encoding is needed so you could also opt to drop `config` here and just pass `uri` and `encoding`.

Reply:
Indeed I don't know how much it can save, but it safe to assume that's preferable to compute the uri only once at sinks init, I will add few words about that and limit the struct member of `DatadogLogsJsonService` (and `DatadogLogsTextService`) to only the mandatory ones.

Reply:
👍 if you swap the fields for only the mandatory ones, I think that would be sufficient. I think the comment is only useful with the current fields where it looks like `uri` is redundant.

Reply:
Unfortunately the `HttpSink.build_request(...)` trait implementation for both `DatadogLogsJsonService` and `DatadogLogsTextService` [calls](https://github.com/timberio/vector/blob/master/src/sinks/datadog/logs.rs#L242) `DatadogLogsConfig.build_request(...)` and use various fields from `DatadogLogsConfig` for request body compression, preventing to easilt remove the `config` field from`DatadogLogsJsonService` and `DatadogLogsTextService`, I think that sticking to the `uri` field with a bit of context seems reasonnable.

Reply:
Ah, I see, I missed that call back into `config`. I'd still be partial to extracting all of the needed fields, like we do in some other sinks, but it isn't a blocker.